src/ORM/Queries/SQLConditionalExpression.php

```diff
@@ -78,7 +78,18 @@ public function addFrom($from)
         if (is_array($from)) {
             $this->from = array_merge($this->from, $from);
         } elseif (!empty($from)) {
-            $this->from[str_replace(['"','`'], '', $from)] = $from;
+            // Make sure we don't have any pointless blank space in our From clause
+            $from = trim($from);
+
+            // Check if the from clause looks like a regular table name
+            // Table name can contain only alphanumeric characters and underscores
+            if (preg_match('/^["`]?([a-zA-Z0-9_]+)["`]?$/', $from)) {
```

Comment:
In theory, you could have other characters in a MySQL table name, but we had a pretty strong assumption over the years that only alphanumeric characters and underscores are acceptable.

Reply:
It's probably safer and easier to just say "if there's whitespace, it's not a valid table name. If there's no whitespace, that's all it _can_ be"
```suggestion
            if (preg_match('/\s/', $from)) {
```

Reply:
That's a good call.

Reply:
Removed the trim and tweaked the regex. I added an extra "padded with spaces" test cases. It's worth nothing that the alias will also be padded with spaces which the previous implementation would have done too.